### PR TITLE
PRO 17:Implement UpdTriggerLeadStatus

### PR DESCRIPTION
**Pull request by GulzatB**

### CODE REVIEW

**Review by belekomurzakov — CHANGES_REQUESTED** [review]

File: force-app/main/default/triggers/UpdTriggerLeadStatus.trigger [hunk 1]
```diff
@@ -0,0 +1,7 @@
+trigger UpdTriggerLeadStatus on Lead (before update) {
```
[the remark this answers is not in the document]
Comment (GulzatB) [reply force-app/main/default/triggers/UpdTriggerLeadStatus.trigger:1]: Got it. Thanks!

File: force-app/main/default/classes/LeadService.cls [hunk 2]
```diff
@@ -1,3 +1,9 @@
 public with sharing class LeadService {
-
+public static void UpdTriggerLeadStatus (list<Lead>newLead){
```
Comment (belekomurzakov) [inline force-app/main/default/classes/LeadService.cls:2]: @GulzatB method name should start with lowerCase, I think updateLeadStatus is better name.

File: force-app/main/default/classes/LeadTriggerHandler.cls [hunk 3]
```diff
@@ -13,6 +13,6 @@ public class LeadTriggerHandler extends TriggerHandler {
     }
 
     public override void beforeUpdate() {
-      
+      LeadService.UpdTriggerLeadStatus(newLeadList);
```
Comment (belekomurzakov) [inline force-app/main/default/classes/LeadTriggerHandler.cls]: don't forget to change name here as well

File: force-app/main/default/classes/LeadService.cls (hunk 2 above)
Comment (belekomurzakov) [reply force-app/main/default/classes/LeadService.cls:2]: also please add Apex Doc above the method, you can check how it's written in ContactService.
add following annotations

@param
@author

**Review by belekomurzakov — CHANGES_REQUESTED** [review]

File: force-app/main/default/triggers/UpdTriggerLeadStatus.trigger (hunk 1 above)
[the remark this answers is not in the document]
Comment (belekomurzakov) [reply force-app/main/default/triggers/UpdTriggerLeadStatus.trigger:1]: delete this old trigger pls. just delete and commit

File: force-app/main/default/classes/LeadTriggerHandler.cls (hunk 3 above)
Comment (GulzatB) [reply force-app/main/default/classes/LeadTriggerHandler.cls]: I did .

File: force-app/main/default/classes/LeadService.cls (hunk 2 above)
Comment (GulzatB) [inline force-app/main/default/classes/LeadService.cls]: Yeah, sounds good :)

**Review by belekomurzakov — APPROVED** [review]
good job!

**Pull request merged**